packages/typescript-to-proptypes/src/createType.ts

```diff
@@ -141,15 +143,7 @@ export function uniqueUnionTypes(node: UnionType): UnionType {
     type: node.type,
     jsDoc: node.jsDoc,
     types: _.uniqBy(node.types, (type) => {
-      if (type.type === 'LiteralNode') {
-        return type.value;
-      }
-
-      if (type.type === 'InstanceOfNode') {
-        return `${type.type}.${type.instance}`;
-      }
-
-      return type.type;
+      return getTypeHash(type);
```

Comment:
Here is the root cause of the issue. If you do the union of two interfaces, they were erased because their `type.type` is `"InterfaceNode"` so only one of the two survive to the `_.uniqBy`

I created a `getTypeHash` that generate a hash in a recursive way to support interface, union and array